crates/ra_hir/src/expr.rs

```diff
@@ -54,6 +54,8 @@ pub struct BodySourceMap {
     pat_map: FxHashMap<PatPtr, PatId>,
     pat_map_back: ArenaMap<PatId, PatPtr>,
     field_map: FxHashMap<(ExprId, usize), AstPtr<ast::NamedField>>,
+    //mapping for StructList exprId to it's name filed list ast node pointer
+    fields_list_map: FxHashMap<ExprId, AstPtr<ast::NamedFieldList>>,
```

Comment:
I think it makes sense to keep source-map smaller. Why do we *need* named_field_list? Can we use the containg `StructLit` instead, which should be present in exprs map?

Reply:
I did not sort out with ast api I guess. So i've made this by analogy with field_map.